File: ARCADE/Intro/Level 9_4: digitDegree.cs
    using System;
     
    public class Test
    {
    	public static int digitDegree(int n){
            int count = 0;
            while (n / 10 > 0){
                n = sumA(n);
                count++;
            }
            return count;
        }
     
        public static int sumA(int n){
            int sum = 0;
            while (n != 0){
                int mod = n % 10;
                sum += mod;
                n /= 10;
            }
            return sum;
        }
    	public static void Main()
    	{
    		int n = 5;
    		Console.WriteLine(digitDegree(n));
    		int m = 91;
    		Console.WriteLine(digitDegree(m));
    	}
    }

File: ARCADE/Intro/Level 9_4: digitDegree.pas
program digitDegree;
 
function sumA(n: integer): integer;
var sum : integer;
begin
    sum := 0;
    while(n > 0) do
    begin
        sum := sum + n mod 10;
        n := n div 10;
    end;
    sumA := sum;
end;
 
function digitDegree(n: integer): integer;
var rs : integer;
begin
    rs := 0;
    while(n >= 10) do
    begin
        n := sumA(n);
        inc(rs);
    end;
    digitDegree := rs;
end;
 
var n : integer;
begin
	n := 5;
	writeln(digitDegree(n));
	n := 100;
	writeln(digitDegree(n));
	n := 91;
	writeln(digitDegree(n));
end.
